internal/typeinfo/typeinfo.go

```diff
@@ -0,0 +1,115 @@
+package typeinfo
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+var cacheMutex sync.RWMutex
+var cache = make(map[reflect.Type]*Info)
+
+// Reflect will return the Info of a given type,
+// generating and caching as required.
+func TypeInfo(value any) (*Info, error) {
+	if value == (any)(nil) {
+		return &Info{}, fmt.Errorf("cannot reflect nil value")
```

Comment:
On errors we pretty much always return the zero value, nil for pointers, unless there are good reasons not to. This is one more way to prevent the error from being ignored, and also prevents a needless allocation as that value is broken and really shouldn't be used.

Reply:
Done everywhere. Thanks.